### ISSUE TITLE
Duplicate SC usage in one app

### ISSUE THREAD

**Issue by allexysleeps** [styled-components/styled-components#1740]
## Reproduction

I have ui-components lib and main app. Both using SC. When I import Component from library. It breaks all App's SC styles. I got warning about duplicate usage.
There are some problems:
1) fix described in [doc](https://www.styled-components.com/docs/faqs#why-am-i-getting-a-warning-about-several-instances-of-module-on-the-page) is depreciated.

2) I've reproduce it on webpack 4 way using [splitChunksPlugin](https://webpack.js.org/plugins/split-chunks-plugin/). It looks like this

```
const rootPath = path.join(__dirname, '../../packages');

module.exports = {
  context: path.join(rootPath),
  entry: {
    vendors: ['styled-components'],
    main: path.join(rootPath, './index.js')
  },
  mode: 'production',
  output: {
    path: path.join(rootPath, '../build/'),
    filename: "[name].bundle.js",
    library: "tp-components",
    libraryTarget: 'umd',
  },
  module: {
    rules: rules
  },
  plugins: [
    ...plugins,
    new UglifyJsPlugin()
  ],
  resolve: {
    extensions: ['.js', '.jsx'],
  },
  externals: {
    'styled-components': {
         commonjs: 'styled-components',
         commonjs2: 'styled-components',
         amd: 'styled-components',
    },
  },
  optimization: {
    splitChunks: {
      chunks: "all",
      minSize: 30000,
      minChunks: Infinity,
      maxAsyncRequests: 5,
      maxInitialRequests: 3,
      automaticNameDelimiter: '~',
      name: true,
      cacheGroups: {
        commons: {
          test: /[\\/]node_modules[\\/]/,
          name: "vendors",
          chunks: "all",
        }
      }
    },
  },
};
```
But it didn't solve issue.

## Expected Behavior

Import Components from lib and keep app styles

## Actual Behavior

App styles is broken

# Environment
## System:
 - OS: macOS High Sierra 10.13.4
 - CPU: x64 Intel(R) Core(TM) i5-5257U CPU @ 2.70GHz
 - Free Memory: 18.72 MB
 - Total Memory: 8.00 GB
 - Shell: /bin/bash - 3.2.57
## Binaries:
 - Node: 8.9.4
 - Yarn: 0.27.5
 - npm: 6.0.1
 - Watchman: 4.7.0
### styled-components:
 - wanted: ^3.2.6
 - installed: 3.2.6

**Comment by probablyup** [comment]
Try adding this to your config:

```js
resolve: {
  alias: {
    'styled-components': require.resolve('styled-components')
  }
}
```

**Comment by gribnoysup** [comment]
Hi @allexysleeps! That is a good catch, we will need to update the docs section to use new `splitChunks` API and not deprecated one 👍

I noticed that you mentioned using your own component library. The solution you tried to use is not the only one and is more suitable for running two separate applications on one page. Based on your Webpack config I would assume that you are probably encountering an issue with several `styled-components` libraries across your `node_modules` dependencies. The docs section mentions this as [another case, with its own solution to the problem](https://www.styled-components.com/docs/faqs#duplicated-module-in-node_modules), that was also mentioned by @probablyup and @epilande, have you tried that one?

**Comment by allexysleeps** [comment]
Hi guys, thank's for your answers.

I've solved this problem by simply changed local lib import.
I used `"my-components-lib": "../../my-lib-root"`
Simply adding `"my-components-lib": "file:../../my-lib-root"`
solved the problem.

However, then I've decided to recompose lib as separate files es5 files instead of one webpack bundle. It looked as better idea for me. And I struggled with the same problem. Then I had to switch to other tasks and left lib as it was as bundle file. When I returned to lib and merged all changes from master. Everything started to work fine. I didn't do any specific changes to project config, so unfortunately I can't provide my solution fo future developers.

I think this issue can be closed. I've mentioned 1 depreciation problem in doc, so this issue wasn't completely useless :)
